crates/optimism/rpc/src/eth/transaction.rs

```diff
@@ -89,3 +93,33 @@ where
         self.sequencer_client.get().cloned()
     }
 }
+
+/// Builds OP transaction response type.
+#[derive(Clone, Debug, Copy)]
+pub struct OpTxBuilder;
+
+impl TransactionCompat for OpTxBuilder {
+    type Transaction = Transaction;
+
+    fn fill(tx: TransactionSignedEcRecovered, tx_info: TransactionInfo) -> Self::Transaction {
+        let signed_tx = tx.clone().into_signed();
+
+        let inner = EthTxBuilder::fill(tx, tx_info).inner;
+
+        Transaction {
+            inner,
+            source_hash: signed_tx.source_hash(),
+            mint: signed_tx.mint(),
+            is_system_tx: signed_tx.is_deposit().then_some(signed_tx.is_system_transaction()),
+            deposit_receipt_version: None, // todo: how to fill this field?
```

Comment:
how are we currently filling this?

Reply:
also with `None`, the field feels a bit off on the tx type, or we need to figure our how to set it, as sep issue

https://github.com/paradigmxyz/reth/blob/c3d090adf43acdb3d9bade49fee9c3ddcfdaed1a/crates/rpc/rpc-types-compat/src/transaction/mod.rs#L101-L111